### PR TITLE
[plugin.video.twit] 2.1.8

### PR DESCRIPTION
**Pull request by divingmule**
### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.twit/changelog.txt [hunk 1]
```diff
@@ -0,0 +1,157 @@
+Version 2.1.8
```
Comment (enen92) [inline plugin.video.twit/changelog.txt]: please drop from submission. Changelog is no longer parsed by kodi in favor of the `<news>` tag

File: plugin.video.twit/addon.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+<addon id="plugin.video.twit"
+             name="TWiT"
+             version="2.1.8"
+             provider-name="divingmule">
+  <requires>
+    <import addon="xbmc.python" version="2.26.0"/>
+    <import addon="script.module.requests" version="2.22.0"/>
+    <import addon="script.module.feedparser" version="5.2.1"/>
+    <import addon="script.ircchat" version="1.0.1" optional="true"/>
+    <import addon="script.module.simple.downloader" version="1.9.5"/>
```
Comment (enen92) [inline plugin.video.twit/addon.xml]: is this module already py3 compatible?

File: plugin.video.twit/default.py [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+        if ret >= 0:
+            resolved_url = list(media_urls.values())[ret]
+    return resolved_url
+
+
+def download_file(stream_url, title):
+    """ thanks/credit to TheCollective for SimpleDownloader module"""
+    path = addon.getSetting('download')
+    if path == "":
+        xbmc.executebuiltin("XBMC.Notification(%s,%s,10000,%s)"
```
Comment (enen92) [inline plugin.video.twit/default.py]: use the python api whenever possible: `xbmcgui.Dialog().notification()`

File: plugin.video.twit/default.py [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+    xbmcplugin.addDirectoryItem(int(sys.argv[1]), plugin_url,
+                                listitem, isfolder)
+
+
+def run_ircchat():
+    # check chat args
+    nickname = addon.getSetting('nickname')
+    username = addon.getSetting('username')
+    if not nickname or not username:
+        xbmc.executebuiltin('XBMC.Notification(%s, %s,10000,%s)' %
```
Comment (enen92) [inline plugin.video.twit/default.py]: same

File: plugin.video.twit/default.py [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+        return
+    # run ircchat script
+    xbmc.executebuiltin(
+        'RunScript(script.ircchat, run_irc=True&nickname=%s&username=%s'
+        '&password=%s&host=irc.twit.tv&channel=twitlive)' %
+        (nickname, username, addon.getSetting('password'))
+    )
+
+
+def set_view_mode():
```
Comment (enen92) [inline plugin.video.twit/default.py]: not allowed in the repo anymore, please remove completely

File: plugin.video.twit/addon.xml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+      https://twit.tv/about/what-is-twit-tv
+      [CR]The TWiT.tv Netcast Network with Leo Laporte features the #1 ranked technology podcast This Week in Tech, along with over 20 other top-ranked online shows.
+    </description>
+    <description lang="no">
+      https://twit.tv/about/what-is-twit-tv
+      [CR]TWiT.tv Netcast-nettverket med Leo Laporte har den topprangerte teknologi-podkasten «This Week in Tech», sammen med over 20 andre topprangerte online show.
+    </description>
+    <platform>all</platform>
+    <language>en no</language>
+    <license>GNU GENERAL PUBLIC LICENSE. Version 2, June 1991</license>
```
Comment (enen92) [inline plugin.video.twit/addon.xml]: please use a SPDX compliant identifier for the license if possible. See: https://spdx.org/licenses/

File: plugin.video.twit/addon.xml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    <import addon="script.module.simple.downloader" version="1.9.5"/>
+    <import addon="script.module.beautifulsoup4" version="4.6.2"/>
+  </requires>
+  <extension point="xbmc.python.pluginsource" library="default.py">
+    <provides>audio video</provides>
+  </extension>
+  <extension point="xbmc.addon.metadata">
+    <summary lang="en">The TWiT.tv Netcast Network</summary>
+    <summary lang="no">TWiT.tv Netcast-nettverk</summary>
+    <description lang="en">
```
Comment (enen92) [inline plugin.video.twit/addon.xml]: en_GB

File: plugin.video.twit/addon.xml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+      https://twit.tv/about/what-is-twit-tv
+      [CR]The TWiT.tv Netcast Network with Leo Laporte features the #1 ranked technology podcast This Week in Tech, along with over 20 other top-ranked online shows.
+    </description>
+    <description lang="no">
```
Comment (enen92) [inline plugin.video.twit/addon.xml]: no_NO ?

File: plugin.video.twit/addon.xml [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    <import addon="script.module.feedparser" version="5.2.1"/>
+    <import addon="script.ircchat" version="1.0.1" optional="true"/>
+    <import addon="script.module.simple.downloader" version="1.9.5"/>
+    <import addon="script.module.beautifulsoup4" version="4.6.2"/>
+  </requires>
+  <extension point="xbmc.python.pluginsource" library="default.py">
+    <provides>audio video</provides>
+  </extension>
+  <extension point="xbmc.addon.metadata">
+    <summary lang="en">The TWiT.tv Netcast Network</summary>
```
Comment (enen92) [inline plugin.video.twit/addon.xml]: en_GB and no_NO

File: plugin.video.twit/addon.xml [hunk 10]
```diff
@@ -0,0 +1,39 @@
+﻿<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="plugin.video.twit"
+             name="TWiT"
+             version="2.1.8"
+             provider-name="divingmule">
+  <requires>
+    <import addon="xbmc.python" version="2.26.0"/>
+    <import addon="script.module.requests" version="2.22.0"/>
+    <import addon="script.module.feedparser" version="5.2.1"/>
```
Comment (enen92) [inline plugin.video.twit/addon.xml]: already py3 compatible?

Comment (divingmule) [reply plugin.video.twit/addon.xml]: Yes, with a small change to the urllib import but, there are dependency issues with the feedparser and the downloader modules with kodi v19.
Should I make fixes and re-submit to the krypton branch?

**Comment by basrieter** [comment]
I am also curious what we should do with https://github.com/xbmc/repo-plugins/tree/gotham/plugin.video.twit, https://github.com/xbmc/repo-plugins/tree/helix/plugin.video.twit? Do we want them marked as broken?

**Comment by enen92** [comment]
ping @divingmule

File: plugin.video.twit/addon.xml (hunk 10 above)
Comment (enen92) [reply plugin.video.twit/addon.xml]: That's fine. Kodi will have a breaking change in Matrix so you'll need to resubmit to matrix at some point

**Comment by divingmule** [comment]
> I am also curious what we should do with https://github.com/xbmc/repo-plugins/tree/gotham/plugin.video.twit, https://github.com/xbmc/repo-plugins/tree/helix/plugin.video.twit? Do we want them marked as broken?

It's fine with me. Do I need to send pull request with <broken> tags?

Thank you all for what you do!

**Comment by enen92** [comment]
If you don't want to support gotham and helix or if you don't mantain those versions anymore, please submit 2 prs marking them as broken. We can do it too but if you can make our life easier, the best ;)

**Comment by enen92** [comment]
The `<broken>` tag is visible to users so try to use a proper message: "This addon is only available for leia and above versions, please upgrade"

**Pull request merged**